libraries/filter_helper.rb

```diff
@@ -46,6 +46,35 @@ def lines_between(current, start, next_match)
       lines || []
     end
 
+    def options(values, allowed)
+      # allowed is {option_name: [settings]}
+      @options ||= {}
+      @options[:safe] ||= safe_default
+      return @options unless values
+      values.each do |key, setting|
+        raise ArgumentError, "Option key  #{key} should be one of #{allowed.keys}" unless allowed.key?(key.to_sym)
+        raise ArgumentError, "Option setting of #{key} should be one of #{allowed[key.to_sym]}" unless allowed[key.to_sym].include?(setting)
+        @options[key.to_sym] = setting
+      end
+      @options
+    end
+
+    def safe_default
+      defined?(@safe) ? @safe : true
+    end
+
+    def safe_default=(option)
+      @safe = option
+    end
```

Comment:
What's the point of this and the default on the resource?

Reply:
This really feels like the wrong way of doing it. The interface should be on the resource, rather than a library. 
Granted it's duplication of code, but that does make it easier for future contributors. 

Right now I can see a future contributor going "why the hell did this person do this?", to this code.

Reply:
I see your point.  Burying a default in the library is odd.  

The filters can be used in tandem inside of a resource and each filter can have a different setting for the safe value. So I need someway to set the default value at the resource level and have an overall default. 

I can yank the default value and then the code returns whatever the resource set. Would that make sense?  Or is the idea of each filter being allowed to override the safe value the problem?

Reply:
@damacus Could I get some more feed back on what you don't like about this, after the changes I made?  No rush, I'm pretty busy.

Reply:
Rather than set an overall default of safe, could we set it at each resource and default them to safe? Then we end up with the same end result right? `== everything defaults to safe, but can be overridden on a per-resource basis`

Reply:
The default for safe is now set in the filter resource.  I rebased this PR.